### PR TITLE
[CDAP-13520] Reports detail bar chart changes

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-13520

Previously, for each bucket, we had two bars next to each other, one bar to show the runs by start method, and the other to show by run status. We could also click on the legends to filter out the runs.

With the new design, we can't filter the runs by each legend anymore. Instead, we show a dropdown to toggle the views for the chart. Selecting the 'Run status' view will show a bar for running and a bar for success/failed runs in each bucket, and selecting 'Start method' view will just show one bar for all the start methods.

This PR also adds highlighting for the currently selected bucket.

Will add legends for the y-axis and other improvements in a later PR.

### CODE REVIEW

**Comment by elfenheart** [comment]
It will be cleaner to create 2 `<rect>`s, one for the styling update at the bottom layer, and another one on top layer to handle the click event. However, because now the `this` context is no longer relevant to the click handler rect, you need a way to select the bottom layer rect. One way I can think of is to assign a unique id to each bucket, then just assign that unique id to the bottom layer rect elements, then you can do a select with that id and just add an active class to it.

**Review by ajainarayanan — COMMENTED** [review]
👍 Code LGTM

File: cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/Legends/Legends.scss [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      width: 150px;
+      margin-left: 10px;
+      border-bottom: 1px solid $grey-01;
+
+      .form-control {
+        border: 0;
+        height: initial;
+        box-shadow: none;
+        cursor: pointer;
+        background-color: $grey-08;
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/Legends/Legends.scss]: these colors should be set as variable for the particular element at the top

File: cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/Legends/index.js [hunk 2]
```diff
@@ -12,136 +12,119 @@
...
+import {ViewByOptions} from 'components/OpsDashboard/store/DashboardStore';
+import ViewByOptionSelector from 'components/OpsDashboard/RunsGraph/Legends/ViewByOptionSelector';
 import T from 'i18n-react';
 
+require('./Legends.scss');
+
 const PREFIX = 'features.OpsDashboard.RunsGraph.Legends';
 
-function LegendsView({onClick, manual, schedule, running, success, failed, delay}) {
+function startMethodLegends() {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/Legends/index.js]: component should start with capital letter

File: cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/graphRenderer.js [hunk 3]
```diff
@@ -154,70 +152,89 @@ export function renderGraph(selector, containerWidth, containerHeight, data, leg
...
+  bottomLayer.append('rect')
+    .attr('class', 'bucket bottom-layer-bucket')
+    .attr('data', (d) => d.time)
+    .attr('width', stepWidth)
+    .attr('x', (d) => x(d.time))
+    .attr('height', height);
+
+  // Since we always select the first bucket by default, highlights it
+  d3.select('.bottom-layer-bucket')
+    .classed('selected', true);
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/graphRenderer.js]: just a heads up, I think @nitinmotgi wants this changed to be last bucket

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Comment by leacd** [comment]
LGTM

**Pull request merged**